ui/src/map/layers/utils/switch-layer-utils.ts

```diff
@@ -203,7 +215,11 @@ export function getJointRenderer(
         },
         [
             (_, coord, ctx, { pixelRatio }) => {
-                ctx.fillStyle = mainJoint ? styles.switchMainJoint : styles.switchJoint;
+                ctx.fillStyle = disabled
+                    ? styles.switchJointDisabled
+                    : mainJoint
+                      ? styles.switchMainJoint
+                      : styles.switchJoint;
```

Comment:
Ikävästi näihinkin alkaa tulla lisää ehtoja, mutta tuon ylemmän refaktoroimisen jälkeen katsoin että vaikka tämä ei ideaalinen olekaan, niin se on oikeastaan ainakin omaan silmääni silti luettavampi näin kuin jos sen purkaisi tuollaiseksi `let-if`-härveliksi. Oispa kotlinin `return if` :pepehands:

Reply:
Joo return if olisi kiva mutta ihan vain funktointi ajaa pitkälti saman, koska siellä voi vain returnata jokaisesta if-haarasta eikä tarvitse lettiä.